core-bundle/src/Controller/FrontendModule/TwoFactorController.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller\FrontendModule;
+
+use Contao\CoreBundle\Exception\RedirectResponseException;
+use Contao\CoreBundle\Security\TwoFactor\Authenticator;
+use Contao\CoreBundle\Translation\Translator;
+use Contao\FrontendUser;
+use Contao\ModuleModel;
+use Contao\PageModel;
+use Contao\Template;
+use ParagonIE\ConstantTime\Base32;
+use Scheb\TwoFactorBundle\Security\Authentication\Exception\InvalidTwoFactorCodeException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorage;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Http\Authentication\AuthenticationUtils;
+
+class TwoFactorController extends AbstractFrontendModuleController
+{
+    /**
+     * @var Translator
+     */
+    protected $translator;
+
+    /**
+     * @var RouterInterface
+     */
+    protected $router;
+
+    /**
+     * @var TokenStorage
+     */
+    protected $tokenStorage;
+
+    /**
+     * @var Authenticator
+     */
+    protected $authenticator;
+
+    /**
+     * @var AuthenticationUtils
+     */
+    protected $authenticationUtils;
+
+    /** @var PageModel */
+    protected $page;
+
+    public function __construct(Translator $translator, RouterInterface $router, TokenStorage $tokenStorage, Authenticator $authenticator, AuthenticationUtils $authenticationUtils)
+    {
+        $this->translator = $translator;
+        $this->router = $router;
+        $this->tokenStorage = $tokenStorage;
+        $this->authenticator = $authenticator;
+        $this->authenticationUtils = $authenticationUtils;
+    }
+
+    public function __invoke(Request $request, ModuleModel $model, string $section, array $classes = null, PageModel $page = null): Response
+    {
+        $this->page = $page;
+
+        if ($this->get('contao.routing.scope_matcher')->isFrontendRequest($request)) {
+            $this->page->loadDetails();
+        }
+
+        return parent::__invoke($request, $model, $section, $classes);
+    }
+
+    protected function getResponse(Template $template, ModuleModel $model, Request $request): Response
+    {
+        $user = $this->tokenStorage->getToken()->getUser();
+
+        if (!$user instanceof FrontendUser) {
+            return $template->getResponse();
+        }
+
+        /** @var PageModel $adapter */
+        $adapter = $this->get('contao.framework')->getAdapter(PageModel::class);
+
+        $redirectPage = $adapter->findByPk($model->jumpTo);
```

Comment:
Maybe check for `$model->jumpTo > 0` to prevent unnecessary DB query? Or is the field mandatory?

Reply:
Fixed in e2ee56b46.